src/core/server/plugins/plugins_service.ts

```diff
@@ -90,12 +96,14 @@ export class PluginsService implements CoreService<PluginsServiceSetup, PluginsS
       .pipe(map((rawConfig) => new PluginsConfig(rawConfig, coreContext.env)));
   }
 
-  public async discover() {
+  public async discover({ environment }: PluginsServiceDiscoverDeps) {
```

Comment:
So, `coreContext` is created synchronously inside the `Server` constructor. Injecting the uuid inside it was way too heavy changes, as it would require to wait for the uuid service to setup (which itself requires the core context...). I'm now setuping the environment service before the call to `plugins.discover` and pass the setup contract to it to allow to access the uuid (and later, the pid)